app/src/main/java/com/example/habithelper/fragments/HomeFragment.java

```diff
@@ -142,6 +174,16 @@ public void done(List<TrackDay> daysTracked, ParseException e) {
                     return;
                 }
                 numDaysTracked = daysTracked.size();
+                currentUser.put("numDaysTracked", numDaysTracked);
+                currentUser.saveInBackground(new SaveCallback() {
+                    @Override
+                    public void done(ParseException e) {
+                        if (e != null) {
```

Comment:
seems like empty body regardless of e value, so you can remove the return statements